com.microsoft.mrtk.input/Utilities/PoseSource/PolyfillHandRayPoseSource.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using System;
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    [Serializable]
+
+    /// <summary>
+    /// A pose source which represents a hand ray. This hand ray is constructed by deriving it from the
+    /// palm and knuckle positions
+    /// </summary>
+    public class PolyfillHandRayPoseSource : HandBasedPoseSource
+    {
+        // The Hand Ray used to calculate the polyfill.
+        private HandRay handRay = new HandRay();
+
+        /// <summary>
+        /// A cache of the knuckle joint pose returned by the hands aggregator.
+        /// </summary>
+        private HandJointPose knuckle;
+
+        /// <summary>
+        /// A cache of the knuckle joint pose returned by the hands aggregator.
+        /// </summary>
+        private HandJointPose palm;
+
+        /// <summary>
+        /// Tries to get the pose of the hand ray by deriving it from the
```

Comment:
May be useful to mention we are returning a pose in world space.

Reply:
+1, always specify reference frame of poses... See above comments about the reference frames in the interactors as well

Reply:
Clarified